server/item/stack.go

```diff
@@ -42,6 +44,26 @@ func NewStack(t world.Item, count int) Stack {
 	return Stack{item: t, count: count, id: newID()}
 }
 
+// WithArmourTrim returns a new stack with the ArmorTrim passed.
+// This only applies if the stack is type Armour.
+func (s Stack) WithArmourTrim(trim ArmourTrim) Stack {
+	if _, ok := s.item.(Armour); !ok {
+		return s
+	}
+
+	s.armourTrim = &trim
+	return s
+}
+
+// ArmourTrim returns the ArmourTrim.
+// if this returns false it does not have an armor trim, or it is not an armour
```

Comment:
Grammar could be better here, maybe "ArmourTrim returns the ArmourTrim and true if present, otherwise false is returned."